tests/unittest_inference.py

```diff
@@ -5312,26 +5304,20 @@ class Cls:
 def test_prevent_recursion_error_in_igetattr_and_context_manager_inference():
     code = """
     class DummyContext(object):
-        def method(self, msg): # pylint: disable=C0103
-            pass
         def __enter__(self):
-            pass
+            return self
         def __exit__(self, ex_type, ex_value, ex_tb):
             return True
 
-    class CallMeMaybe(object):
-        def __call__(self):
-            while False:
-                with DummyContext() as con:
-                    f_method = con.method
-                break
+    if False:
+        with DummyContext() as con:
+            pass
 
-            with DummyContext() as con:
-                con #@
-                f_method = con.method
+    with DummyContext() as con:
+        con.__enter__  #@
     """
     node = extract_node(code)
-    assert next(node.infer()) is util.Uninferable
+    next(node.infer())  # should not raise StopIteration/RuntimeError
```

Comment:
This test is unrelated to the inference change, but failed because `con` went from `Uninferable` to `None` (i.e. the behaviour tested was over-specified). There is some additional explanation in the commit message fixing this but the short story is that while fixing it I noticed that it didn't actually test the regression (i.e. it passed with and without the fix in 55076ca)

Reply:
How about I move it to another MR?

> Thus the last line could be assert isinstance(next(node.infer()), Const) and assert (next(node.infer()).value == None). Am i wrong?

You're not wrong per se (though it would be `BoundMethod __enter__ of DummyContext` if I'm reading it correctly), but I didn't change that for the same reason that I detailed above: the test is *actually* testing (in my opinion), that inference of these nodes does not crash, not what you're actually inferring them as. If you over-specify the behaviour of tests, they have a tendency to be "brittle", or break when unrelated changes occur, just like this test in the first place.

Reply:
@nelfin you are right. Maybe you can add a docstring to detail that it test the non-crash behavior and not the result of the inference. What do you think about it?

Reply:
:+1: